### PR TITLE
feat: new event payments.cancel_pre_purchase_reservation

### PR DESCRIPTION
**Pull request by PatricioPoncini**
### `Changes`
new event payments.cancel_pre_purchase_reservation

### `docs`
new event payments.cancel_pre_purchase_reservation. this event is to cancel the pre-purchase reservation of a resource in room-inventory. when the Redis key in payments expires, it will be removed, and consequently, the pre-purchase reservation will be canceled

### CODE REVIEW

**Review by jym272 — CHANGES_REQUESTED** [review]

File: packages/legend-transac/src/@types/event/events.ts [hunk 1]
```diff
@@ -48,6 +48,14 @@ export interface EventPayload {
         userId: string;
         userToUnblockId: string;
     };
+    /**
+     * Event to cancel a pre-purchase reservation for a resource in room-inventory
+     */
+    'payments.cancel_pre_purchase_reservation': {
```
[the remark this answers is not in the document]
Comment (PatricioPoncini) [reply packages/legend-transac/src/@types/event/events.ts:54]: added

Comment (jym272) [inline packages/legend-transac/src/@types/event/events.ts:54]: alphabetical order

File: packages/legend-transac/src/@types/event/events.ts [hunk 2]
```diff
@@ -59,7 +67,8 @@ export const microserviceEvent = {
     'PAYMENTS.NOTIFY_CLIENT': 'payments.notify_client',
     'SOCIAL.BLOCK_CHAT': 'social.block_chat',
     'SOCIAL.NEW_USER': 'social.new_user',
-    'SOCIAL.UNBLOCK_CHAT': 'social.unblock_chat'
+    'SOCIAL.UNBLOCK_CHAT': 'social.unblock_chat',
+    'PAYMENTS.CANCEL_PRE_PURCHASE_RESERVATION': 'payments.cancel_pre_purchase_reservation'
```
Comment (jym272) [inline packages/legend-transac/src/@types/event/events.ts:71]: alphabetical order

**Review by jym272 — CHANGES_REQUESTED** [review]

File: packages/legend-transac/src/@types/event/events.ts [hunk 3]
```diff
@@ -53,6 +61,7 @@ export interface EventPayload {
  * Represents the available events in the system.
  */
 export const microserviceEvent = {
+    'PAYMENTS.CANCEL_PRE_PURCHASE_RESERVATION': 'payments.cancel_pre_purchase_reservation',
```
Comment (jym272) [inline packages/legend-transac/src/@types/event/events.ts:64]: This is the section of the test-events, place this event below this section, after:
```text
/////////////////////////
```

**Review by jym272 — CHANGES_REQUESTED** [review]

File: packages/legend-transac/src/@types/event/events.ts (hunk 3 above)
Comment (PatricioPoncini) [reply packages/legend-transac/src/@types/event/events.ts]: changed

**Pull request merged**